src/meta/src/controller/streaming_job.rs

```diff
@@ -923,6 +925,54 @@ impl CatalogController {
                         )),
                     });
                 }
+
+                // If the index is created on a table with privileges, we should also
```

Comment:
Cc @chenzl25 , fixed here. If there is a default `SELECT` privilege for the tables under the schema, the `SELECT` privilege will be automatically granted when some table is created. After that, when an index is created, the `SELECT` privilege will also be automatically granted.

Reply:
The issue stems from a bug in how index's table privileges are granted and revoked. The access privileges for any newly created index must always mirror those of its primary table. Therefore, even when default privileges are altered post-table-creation, newly generated indexes should retain the same accessibility constraints as the table had at creation time.